### DISCUSSION TITLE
DOcumentation Issue - Probably no applicable document definition builder was found

### DISCUSSION THREAD

**Discussion by bajizs** [dachcom-digital/pimcore-dynamic-search-index-provider-elasticsearch#25] [General]
| Q                | A
| ---------------- | -----
| Bug report?      | no
| Feature request? | yes
| BC Break report? | no
| RFC?             | no

I have set up evreything by Documentation (https://github.com/dachcom-digital/pimcore-dynamic-search/tree/master/docs) and README. With Lucene system works but with elastics I get always next error:

No document definition generated. Probably no applicable document definition builder was found. Skipping.... (The current context index provider also requires pre-configured indices. Please make sure your document definition implements the "DynamicSearchBundle\Provider\PreConfiguredIndexProviderInterface" interface)

Can't found any information what is wrong, where or how to create/configure PreConfiguredIndexProviderInterface. I have added my index configuration class - this works with lucene.

Where can I found any information or example for usage of this bundle?

Thank You

**Comment by raphakat** [comment]
Hi, I have a very similar problem:
I have created/configured elasticsearch as described in [https://github.com/dachcom-digital/pimcore-dynamic-search/blob/master/docs/02_MultiSiteMultiLocaleSetup.md](https://github.com/dachcom-digital/pimcore-dynamic-search/blob/master/docs/02_MultiSiteMultiLocaleSetup.md). However, something seems to be missing in the documentation, which leads to two problems for me:

1. when I run the command **bin/console dynamic-search:run**, **$normalizerOptions['locale']** cannot be accessed in the class **App\DynamicSearch\Definition\Trinity\AssetDefinition.php** and **App\DynamicSearch\Definition\Trinity\CarDefinition.php** in their **buildDefinition(..)**. If I output the passed array **$normalizerOptions**, it is always empty (**buildDefinition(..)** is called in **DynamicSearchBundle\Manager\DocumentDefinitionManager->generateDocumentDefinitionForContext(..)** with an empty array). 
As a workaround to get the whole thing working, I set **$normalizerOptions['locale'] = \Pimcore\Tool::getDefaultLanguage()** in the **buildDefinition(..)**. Is there a specific configuration, event or class missing in the previous documentation that still needs to be adjusted so that $normalizerOptions is filled?

2. when I used the workaround in 1. to get the whole thing to work without errors, the indexes are not created, which makes the search impossible.
Is there still a piece of the puzzle missing in the documentation or do you have an idea why this could be?

I am using Pimcore 11, [https://github.com/ddev/ddev-elasticsearch](https://github.com/ddev/ddev-elasticsearch) with the default configuration and [https://github.com/JanoPL/ddev-kibana](https://github.com/JanoPL/ddev-kibana) as UI to display the indexes.

Thank you

**Comment by solverat** [comment]
@bajizs are you using `PreProcessFieldDefinitions`? Note, that these are not supported for ES, those only works with lucene (see https://github.com/dachcom-digital/pimcore-dynamic-search/discussions/85). 

Also, the `index_transformer` config node in `addSimpleDocumentFieldDefinition` are slightly different. A basic ES index field example would look like this:

```php
->addSimpleDocumentFieldDefinition([
    'name'              => 'type',
    'index_transformer' => [
        'type'          => 'explicit',
        'configuration' => [
            'type' => 'keyword',
            'ignore_above' => 256,
        ]
    ],
    'data_transformer'  => [
        'type'          => MyExtractor::class,
    ]
])
```
